frontend/src/APIClients/SchedulingAPIClient.ts

```diff
@@ -0,0 +1,98 @@
+import AUTHENTICATED_USER_KEY from "../constants/AuthConstants";
+import { Schedule, UpdatedSchedulingFields } from "../types/SchedulingTypes";
+import { getLocalStorageObjProperty } from "../utils/LocalStorageUtils";
+import baseAPIClient from "./BaseAPIClient";
+
+const getSchedules = async (): Promise<Schedule[]> => {
+  const bearerToken = `Bearer ${getLocalStorageObjProperty(
+    AUTHENTICATED_USER_KEY,
+    "accessToken",
+  )}`;
+  try {
+    const { data } = await baseAPIClient.get("/scheduling", {
+      headers: { Authorization: bearerToken },
+    });
+    return data;
+  } catch (error) {
+    return [];
+  }
+};
+
+const getScheduleById = async (scheduleId: string): Promise<Schedule> => {
+  const bearerToken = `Bearer ${getLocalStorageObjProperty(
+    AUTHENTICATED_USER_KEY,
+    "accessToken",
+  )}`;
+  try {
+    const { data } = await baseAPIClient.get(`/scheduling/${scheduleId}`, {
+      headers: { Authorization: bearerToken },
+    });
+    return data;
+  } catch (error) {
+    return null;
+  }
+};
+
+const createSchedule = async (schedule: Schedule): Promise<Schedule> => {
```

Comment:
Appears that I still need to accommodate for foreign key reference
![Screen Shot 2021-11-06 at 4 35 12 PM](https://user-images.githubusercontent.com/38363056/140630038-ca9984a3-aae1-4d4e-a86a-28db18fb92ff.png)

Reply:
This is fine! In fact this is how it should be. To create a donation, you need to to be logged in and that donor should be in the donors table.

Reply:
On a semi-related note, do we still allow the `donorId` property to be updated when we update a schedule? Or do we keep it so that the donorId stays consistent with whichever donor is logged in so they can only edit their scheduled drop-offs?

Reply:
`donorId` should not be updated and the Frontend will never pass `donorId` to the updateSchedule function / endpoint.